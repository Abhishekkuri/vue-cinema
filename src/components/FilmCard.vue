<template>
  <div class="card-container">
    <div class="card">
      <div class="poster">
        <img :src="film.Poster" alt="Poster Not Available" />
      </div>
      <div class="card-detail">
        <h4 class="card-detail-item">{{ film.Title }}</h4>
        <div class="card-detail-item">Type: {{ film.Type }}</div>
        <div class="card-detail-item">Rated: {{ film.Rated }}</div>
        <div card-detail-item>
          Genre:
          <span
            class="custom-span"
            v-for="genre in film.Genre.split(',')"
            :key="genre"
          >
            {{ genre }}
          </span>
        </div>
        <div class="card-detail-item">Year: {{ film.Year }}</div>
        <div class="card-detail-item">
          Actors:
          <span
            class="custom-span actor"
            v-for="actor in film.Actors.split(',')"
            :key="actor"
          >
            {{ actor }}
          </span>
        </div>
        <div class="card-detail-item">Directors: {{ film.Director }}</div>
        <div class="card-detail-item">Awards: {{ film.Awards }}</div>
        <p class="card-detail-item">
          Plot:
          <span>{{ filmPlot }}</span>
          <a
            class="read-more"
            v-if="readMoreActivated && film.Plot.length > 200"
            @click.prevent="toggleReadMore"
          >
            read more...
          </a>
          <a
            class="read-more"
            v-if="!readMoreActivated && film.Plot.length > 200"
            @click.prevent="toggleReadMore"
          >
            read less
          </a>
        </p>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  props: ["film"],
  data() {
    return {
      readMoreActivated: this.film.Plot.length > 200,
    };
  },
  computed: {
    filmPlot() {
      return this.readMoreActivated
        ? this.film.Plot.slice(0, 200)
        : this.film.Plot;
    },
  },
  methods: {
    toggleReadMore() {
      this.readMoreActivated = !this.readMoreActivated;
    },
  },
};
</script>

<style lang="scss" scoped>
.card-container {
  padding: 1rem;
  margin: 1rem;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  background-color: $color-secondary;
  &:hover {
    box-shadow: 0 8px 16px 0 $color-primary;
  }
  .poster {
    padding: 5px;
    height: 28rem;
    img {
      width: 100%;
      height: 28rem;
      border-radius: $color-grey-dark;
    }
  }
  .card-detail {
    padding: 5px;
    .card-detail-item {
      padding: 3px 0;
      .read-more {
        text-decoration: none;
        cursor: pointer;
        color: $color-secondary;
      }
    }
  }
  .custom-span {
    border: 1px solid $color-grey-dark;
    display: inline-block;
    margin-right: 5px;
    margin-bottom: 5px;
    padding: 0 3px;
    border-radius: 5px;
  }
}

@media screen and (max-width: 820px) {
}
</style>
