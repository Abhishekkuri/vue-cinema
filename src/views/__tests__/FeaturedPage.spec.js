import { createLocalVue, shallowMount } from "@vue/test-utils";
import Vuex from "vuex";
import FeaturePage from "@/views/FeaturePage";
import FilmCard from "@/components/FilmCard";
import merge from "lodash.merge";
const localVue = createLocalVue();

localVue.use(Vuex);

function createFeaturedFilms() {
  const arr = new Array(2);
  return arr.fill().map((film, i) => ({
    Title: `a${i}`,
    Year: "2022",
    Genre: "Series",
    imdbID: `a${i}`,
  }));
}

function createStore(overrides) {
  const defaultStoreConfig = {
    getters: {
      getFeaturedFilms: jest.fn(() => createFeaturedFilms()),
    },
    actions: {
      fetchFeaturedFilms: jest.fn(() => Promise.resolve()),
    },
  };
  return new Vuex.Store(merge(defaultStoreConfig, overrides));
}

function createWrapper(overrides) {
  const defaultMountingOptions = {
    localVue,
    store: createStore(),
  };
  return shallowMount(FeaturePage, merge(defaultMountingOptions, overrides));
}

describe("FeaturePage.vue", () => {
  test("Featured Page renders correctly", () => {
    expect(createWrapper().element).toMatchSnapshot();
  });
  test("Featured Page show 2 films", () => {
    const wrapper = createWrapper();
    expect(wrapper.findComponent(FilmCard).exists()).toBe(true);
    expect(wrapper.findAllComponents(FilmCard).length).toBe(2);
  });
});
