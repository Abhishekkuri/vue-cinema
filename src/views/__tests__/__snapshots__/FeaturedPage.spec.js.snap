// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturePage.vue Featured Page renders correctly 1`] = `
<div>
  <h1
    class="view-heading"
  >
    Featured Films
  </h1>
   
  <div
    class="featured-container"
  >
    <film-card-stub
      film="[object Object]"
    />
    <film-card-stub
      film="[object Object]"
    />
  </div>
</div>
`;
